From 70a12185ecb780a1cf523d28ed4128ae67b341bd Mon Sep 17 00:00:00 2001
From: Stanley Tng <stng@google.com>
Date: Wed, 10 Oct 2018 18:48:33 -0700
Subject: [PATCH 39/80] Add transaction code to TimeCheck crash

When there is a TimeCheck timeout crash, the transaction code will be
useful for debugging.

Test: Manual run
Bug: 117349352
Change-Id: I8a93ead7ec7379f51b211ca31c68ec6ec2553afc
Merged-In: I4716852c77d56703ad5f3dfc2500f598a2b80a12
(cherry picked from commit de9e33dfcbdabc0227ab47e4a13325b5b410ab39)
---
 media/libaudioclient/IAudioFlinger.cpp       | 4 +++-
 media/libaudioclient/IAudioPolicyService.cpp | 4 +++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/media/libaudioclient/IAudioFlinger.cpp b/media/libaudioclient/IAudioFlinger.cpp
index 00af7e8ea..9f3b74241 100644
--- a/media/libaudioclient/IAudioFlinger.cpp
+++ b/media/libaudioclient/IAudioFlinger.cpp
@@ -951,7 +951,9 @@ status_t BnAudioFlinger::onTransact(
             break;
     }
 
-    TimeCheck check("IAudioFlinger");
+    char timeCheckString[64];
+    snprintf(timeCheckString, sizeof(timeCheckString), "IAudioFlinger: %d", code);
+    TimeCheck check(timeCheckString);
 
     switch (code) {
         case CREATE_TRACK: {
diff --git a/media/libaudioclient/IAudioPolicyService.cpp b/media/libaudioclient/IAudioPolicyService.cpp
index a1236e738..8cd4a8579 100644
--- a/media/libaudioclient/IAudioPolicyService.cpp
+++ b/media/libaudioclient/IAudioPolicyService.cpp
@@ -948,7 +948,9 @@ status_t BnAudioPolicyService::onTransact(
             break;
     }
 
-    TimeCheck check("IAudioPolicyService");
+    char timeCheckString[64];
+    snprintf(timeCheckString, sizeof(timeCheckString), "IAudioPolicyService: %d", code);
+    TimeCheck check(timeCheckString);
 
     switch (code) {
         case SET_DEVICE_CONNECTION_STATE: {
-- 
2.17.1

